lib/Sema/TypeCheckMacros.cpp

```diff
@@ -1568,6 +1581,142 @@ static SourceFile *evaluateAttachedMacro(MacroDecl *macro, Decl *attachedTo,
   return macroSourceFile;
 }
 
+static SourceFile *evaluateAttachedMacro(MacroDecl *macro,
+                                         ClosureExpr *attachedTo,
```

Comment:
Much of this code is duplicated from the other overload of `evaluateAttachedMacro` that accepts a `Decl *`, but it's probably not hard to generalize using `ASTNode` instead.